wait_for_storage_pods() is not ignoring rook-ceph-detect-version pod while checking for all OCS pods state

It looks like wait_for_storage_pods function is not ignoring  rook-ceph-detect-version pod  that we are handling here 
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/pod.py#L1407-L1412

Output captured just after the above mentioned lines of code,
```
14:32:19 - MainThread - ocs_ci.ocs.resources.pod - INFO - -------['csi-cephfsplugin-76t2m', 'csi-cephfsplugin-provisioner-7d4fc9c448-7sqvj', 'csi-cephfsplugin-provisioner-7d4fc9c448-vrk45', 'csi-cephfsplugin-qlr7q', 'csi-cephfsplugin-wz2mb', 'csi-rbdplugin-27dhj', 'csi-rbdplugin-bph6d', 'csi-rbdplugin-provisioner-864bcd5c8f-fbkjc', 'csi-rbdplugin-provisioner-864bcd5c8f-tfh69', 'csi-rbdplugin-rm5k4', 'noobaa-core-0', 'noobaa-db-pg-0', 'noobaa-endpoint-7d447588ff-rm4g5', 'noobaa-operator-d7458fb89-9swfv', 'ocs-metrics-exporter-d84c68cc6-fxwxs', 'ocs-operator-776b6d9f9f-bfcfk', 'rook-ceph-crashcollector-compute-0-778cc99cb4-2pglr', 'rook-ceph-crashcollector-compute-1-5dbbdd9db7-tbdss', 'rook-ceph-crashcollector-compute-2-7b68476d86-jqk4j', **'rook-ceph-detect-version--1-cx82f'**, 'rook-ceph-mds-ocs-storagecluster-cephfilesystem-a-5744c666p65k2', 'rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-98bc98ccclpnf', 'rook-ceph-mgr-a-78fcdf97f4-89b9d', 'rook-ceph-mon-a-545c6d86f8-kz855', 'rook-ceph-mon-b-57b5476fb-mndnd', 'rook-ceph-mon-c-86ddccc674-6k7bx', 'rook-ceph-operator-85558b848c-2cpgz', 'rook-ceph-osd-0-56fd6457ff-pbcrj', 'rook-ceph-osd-1-c69b4c9bd-grjhg', 'rook-ceph-osd-2-76bd877677-p74g6', 'rook-ceph-rgw-ocs-storagecluster-cephobjectstore-a-676f7ffbxh5m', 'rook-ceph-tools-7695784b6b-ptmbj']-----------
```

Error:
```
            )
E           ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc -n openshift-storage describe Pod rook-ceph-detect-version--1-cx82f.
E           Error is Error from server (NotFound): pods "rook-ceph-detect-version--1-cx82f" not found

ocs_ci/utility/utils.py:511: CommandFailed
```

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -23,7 +23,6 @@ from ocs_ci.framework import config
 from ocs_ci.ocs.exceptions import (
     CommandFailed,
     NonUpgradedImagesFoundError,
-    ResourceWrongStatusException,
     TimeoutExpiredError,
     UnavailableResourceException,
     ResourceNotFoundError,
@@ -1406,7 +1405,8 @@ def wait_for_storage_pods(timeout=200):
         pod
         for pod in all_pod_obj
         if pod.get_labels()
-        and constants.ROOK_CEPH_DETECT_VERSION_LABEL[4:] not in pod.get_labels().values()
+        and constants.ROOK_CEPH_DETECT_VERSION_LABEL[4:]
+        not in pod.get_labels().values()
     ]
 
     for pod_obj in all_pod_obj:

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -522,7 +522,9 @@ def get_all_pods(
 
     """
     ocp_pod_obj = OCP(
-        kind=constants.POD, namespace=namespace, field_selector=field_selector,
+        kind=constants.POD,
+        namespace=namespace,
+        field_selector=field_selector,
     )
     # In case of >4 worker nodes node failures automatic failover of pods to
     # other nodes will happen.
@@ -1432,7 +1434,7 @@ def verify_pods_upgraded(old_images, selector, count=1, timeout=720):
     """
 
     namespace = config.ENV_DATA["cluster_namespace"]
-    pod = OCP(kind=constants.POD, namespace=namespace,)
+    pod = OCP(kind=constants.POD, namespace=namespace)
     info_message = (
         f"Waiting for {count} pods with selector: {selector} to be running "
         f"and upgraded."
@@ -1719,7 +1721,10 @@ def get_osd_removal_pod_name(osd_id, timeout=60):
 
     try:
         for osd_removal_pod_names in TimeoutSampler(
-            timeout=timeout, sleep=5, func=get_pod_name_by_pattern, pattern=pattern,
+            timeout=timeout,
+            sleep=5,
+            func=get_pod_name_by_pattern,
+            pattern=pattern,
         ):
             if osd_removal_pod_names:
                 osd_removal_pod_name = osd_removal_pod_names[0]

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -1406,7 +1406,7 @@ def wait_for_storage_pods(timeout=200):
         pod
         for pod in all_pod_obj
         if pod.get_labels()
-        and constants.ROOK_CEPH_DETECT_VERSION_LABEL[4:] not in pod.get_labels().value
+        and constants.ROOK_CEPH_DETECT_VERSION_LABEL[4:] not in pod.get_labels().values()
     ]
 
     for pod_obj in all_pod_obj:

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -523,9 +523,7 @@ def get_all_pods(
 
     """
     ocp_pod_obj = OCP(
-        kind=constants.POD,
-        namespace=namespace,
-        field_selector=field_selector,
+        kind=constants.POD, namespace=namespace, field_selector=field_selector,
     )
     # In case of >4 worker nodes node failures automatic failover of pods to
     # other nodes will happen.
@@ -1408,33 +1406,14 @@ def wait_for_storage_pods(timeout=200):
         pod
         for pod in all_pod_obj
         if pod.get_labels()
-        and constants.ROOK_CEPH_DETECT_VERSION_LABEL not in pod.get_labels()
+        and constants.ROOK_CEPH_DETECT_VERSION_LABEL[4:] not in pod.get_labels().value
     ]
 
     for pod_obj in all_pod_obj:
         state = constants.STATUS_RUNNING
         if any(i in pod_obj.name for i in ["-1-deploy", "ocs-deviceset"]):
             state = constants.STATUS_COMPLETED
-
-        try:
-            helpers.wait_for_resource_state(
-                resource=pod_obj, state=state, timeout=timeout
-            )
-        except ResourceWrongStatusException:
-            # 'rook-ceph-crashcollector' on the failed node stucks at
-            # pending state. BZ 1810014 tracks it.
-            # Ignoring 'rook-ceph-crashcollector' pod health check as
-            # WA and deleting its deployment so that the pod
-            # disappears. Will revert this WA once the BZ is fixed
-            if "rook-ceph-crashcollector" in pod_obj.name:
-                ocp_obj = ocp.OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
-                pod_name = pod_obj.name
-                deployment_name = "-".join(pod_name.split("-")[:-2])
-                command = f"delete deployment {deployment_name}"
-                ocp_obj.exec_oc_cmd(command=command)
-                logger.info(f"Deleted deployment for pod {pod_obj.name}")
-            else:
-                raise
+        helpers.wait_for_resource_state(resource=pod_obj, state=state, timeout=timeout)
 
 
 def verify_pods_upgraded(old_images, selector, count=1, timeout=720):
@@ -1453,10 +1432,7 @@ def verify_pods_upgraded(old_images, selector, count=1, timeout=720):
     """
 
     namespace = config.ENV_DATA["cluster_namespace"]
-    pod = OCP(
-        kind=constants.POD,
-        namespace=namespace,
-    )
+    pod = OCP(kind=constants.POD, namespace=namespace,)
     info_message = (
         f"Waiting for {count} pods with selector: {selector} to be running "
         f"and upgraded."
@@ -1743,10 +1719,7 @@ def get_osd_removal_pod_name(osd_id, timeout=60):
 
     try:
         for osd_removal_pod_names in TimeoutSampler(
-            timeout=timeout,
-            sleep=5,
-            func=get_pod_name_by_pattern,
-            pattern=pattern,
+            timeout=timeout, sleep=5, func=get_pod_name_by_pattern, pattern=pattern,
         ):
             if osd_removal_pod_names:
                 osd_removal_pod_name = osd_removal_pod_names[0]

